source/AI.cpp

```diff
@@ -497,7 +497,7 @@ void AI::ClearOrders()
 
 
 
-void AI::Step(const PlayerInfo &player, Command &activeCommands)
+void AI::Step(const PlayerInfo &player, Command &activeCommands, Angle mouseAngle, bool rightMouseButtonHeld)
```

Comment:
I do not know much about commands but can't the moveAngle and the rightMouseButtonHeld be added to the command?
rightMouseButtonHeld is just to know if we are shooting, right? The idea of the command is that the game engine parses meaning out of whatever the player is typing (not only but it's one thing)

Reply:
Well, I also felt that what I did was not the best way to do it... But couldn't work out a better way. @quyykk advised to avoid calling SDL methods in the AI.cpp itself. And what I saw from existing commands, none of them handles the mouse related things. Also, the other mouse related operations like drag, click, etc are hard coded in the events file. I too wanted to make the system more dynamic so that it will easy in the future to map mouse keys, etc. If you can show me a better way I will be glad to do it in your way.

Reply:
Now onto this one... Yeah thanks for that advice. I think that the commands infrastructure will help me on this.

Reply:
I have made it, so that it uses the PRIMARY command but there is a big problem. The primary command is a general command for the ship to fire its primary weapons... there is no way I can tell if that command is generated after a mouse input or keyboard. So, basically it ain't possible to make the weapons fire with right mouse button only if mouse turning is enabled without using any extra flags.

Reply:
You don't need to know how it was generated; AI.cpp should just executes commands when its possible, in my opinion.
That means Engine.cpp knows if its mouse or keyboard and if its mouse and we're in alt mode then it gives a primary command, otherwise not.

Reply:
@Hurleveur would it make sense to support `Angle` within `PlayerInfo &player`?  The right click held can set a firing command instead of being passed as a step.  If both of these are done then the Step arguments can be reverted.

For example,

```cpp
player.GetMouseAngle()
```